### ISSUE TITLE
unusual water behaviour with gromacs files

### ISSUE THREAD

**Issue by rowanwalker96** [openmm/openmm#4184]
Hi there,

I'm having trouble with a simulation starting from some gromacs files, with topology of each kind of molecule in .itp files, from charmm-gui webserver (attached)
[examplefiles.zip](https://github.com/openmm/openmm/files/12339314/examplefiles.zip)

After minimisation in openmm all water molecules appear broken (see attached image), it looks like the water bonds aren't read correctly from TIP3.itp. top.topology.getNumBonds()=440 which is just the number of bonds of the solute. 

![example (1)](https://github.com/openmm/openmm/assets/60986121/bb0a62ec-c282-4338-aa43-48dbcf7ccf61)

I also found along the way that applying some positional restraints to the heavy atoms of the solute with the below code had no observable effect on the dynamics even at high k which is a bit strange.

```
restraint = CustomExternalForce('k*periodicdistance(x, y, z, x0, y0, z0)^2')
restraint.usesPeriodicBoundaryConditions()
system.addForce(restraint)
restraint.addGlobalParameter('k', 1000.0*kilojoules_per_mole/nanometer**2)
restraint.addPerParticleParameter('x0')
restraint.addPerParticleParameter('y0')
restraint.addPerParticleParameter('z0')

for atom in top.topology.atoms():
    if system.getParticleMass(atom.index)._value > 1.5 and atom.name != 'O' and atom.name != 'SOD': # heavy atoms
        restraint.addParticle(atom.index, gro.positions[atom.index])
```

**Comment by rowanwalker96** [comment]
sorry, the TIP3.itp file from charmmgui did not explicitly include bond information, putting them in everything now looks good

**Comment by sef43** [comment]
For the second part of your issue make sure you are calling `simulation.context.reinitialize` after adding the restraining force. See how it is used in this [cookbook example](https://openmm.github.io/openmm-cookbook/latest/notebooks/cookbook/Changing%20Temperature%20and%20Pressure.html) and in the API: http://docs.openmm.org/latest/api-python/generated/openmm.openmm.Context.html#openmm.openmm.Context.reinitialize

**Comment by rowanwalker96** [comment]
thanks!

**Comment by peastman** [comment]
Glad it was that easy!

Is it ok to close this issue, or are there parts of your question that haven't been answered?

**Comment by rowanwalker96** [comment]
oops sorry, yes here it is
[examplefiles 2.zip](https://github.com/openmm/openmm/files/12352848/examplefiles.2.zip)

**Comment by peastman** [comment]
Do you have any idea what the correct way of interpreting the `[ settles ]` directive is?  The [documentation](https://manual.gromacs.org/current/reference-manual/topologies/constraint-algorithm-section.html) isn't especially helpful.  Here is the complete description: "The [ settles ] directive defines the first atom of the water molecule. The settle funct is always 1, and the distance between O-H and H-H distances must be given."

Is it assumed the first atom will always be the oxygen, and it will be immediately followed by the two hydrogens?  Are we supposed to look at the names and assume the oxygen name will begin with O and the hydrogen names will begin with H?

**Comment by rowanwalker96** [comment]
Apparently [settles] is hard coded so that the order of the atoms should be O H H or O H H Vrt

**Comment by peastman** [comment]
Thanks!

**Comment by peastman** [comment]
The fix is in #4188.

**Comment by rowanwalker96** [comment]
thank you!
